persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -80,16 +81,13 @@ DatabaseType getDatabaseType() {
    * @param scriptFilePath : Path of SQL script.
    * @throws SQLException : Exception while executing the script.
    */
-  public void executeScript(String scriptFilePath) throws SQLException {
-    ClassLoader classLoader = DatasourceOperations.class.getClassLoader();
+  public void executeScript(InputStream scriptInputStream) throws SQLException {
     runWithinTransaction(
         connection -> {
           try (Statement statement = connection.createStatement()) {
             BufferedReader reader =
                 new BufferedReader(
-                    new InputStreamReader(
-                        Objects.requireNonNull(classLoader.getResourceAsStream(scriptFilePath)),
-                        UTF_8));
+                    new InputStreamReader(Objects.requireNonNull(scriptInputStream), UTF_8));
```

Comment:
who is closing `scriptInputStream`?

Reply:
Who was closing it before? When the JVM exits, I suppose the OS will reclaim the resources eventually... and this is a short-lived bootstrap invocation, not the long-running service. But for now, maybe we can just add a `close` in this method and shift the responsibility to close here.

Reply:
I did not look into the previous revision of this code in details :sweat_smile:  but in general, I believe theses streams need to be closed... WDYT?

Reply:
That sounds like a good idea... is this [the one from jsr305](https://www.javadoc.io/static/com.google.code.findbugs/jsr305/3.0.2/javax/annotation/WillClose.html)? It doesn't seem that we have that dependency currently, and we're not using it anywhere else in the project.

Reply:
Oops sorry mixed up the threads.

They don't really _need_ to be closed, but it's definitely a good idea to close them like the code now does.